cpp/src/io/parquet/page_enc.cu

```diff
@@ -1236,22 +1237,23 @@ __global__ void __launch_bounds__(128, 8)
               }
             }
 
-            auto const ret = convert_nanoseconds([&]() {
+            auto const& [last_day_nanos, julian_days] = ([&]() {
+              using namespace cuda::std::chrono;
               switch (s->col.leaf_column->type().id()) {
                 case type_id::TIMESTAMP_SECONDS:
                 case type_id::TIMESTAMP_MILLISECONDS: {
-                  return timestamp_ns{duration_ms{v}};
+                  return julian_days_with_time<cuda::std::milli>(v);
                 } break;
                 case type_id::TIMESTAMP_MICROSECONDS:
                 case type_id::TIMESTAMP_NANOSECONDS: {
-                  return timestamp_ns{duration_us{v}};
+                  return juilian_days_with_time<cuda::std::micro>(v);
```

Comment:
Sorry -- I didn't fix every case in my previous review. I said "and similarly below" but that might have been missed.
```suggestion
                  return julian_days_with_time<cuda::std::micro>(v);
```